File: codesanook-examples-java/src/main/java/com/codesanook/examples/stringinterpolation/Product.java
package com.codesanook.examples.stringinterpolation;

import java.math.BigDecimal;
import java.util.Date;

public class Product {

    private String name;
    private Date createDate;
    private BigDecimal price;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public BigDecimal getPrice() {
        return price;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }
}

File: codesanook-examples-typescript/src/models/Product.ts
export default class Product {

    private _name: string;
    private _createDate: Date;
    private _price: number;

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get createDate(): Date {
        return this._createDate;
    }

    set createDate(value: Date) {
        this._createDate = value;
    }

    get price(): number {
        return this._price;
    }

    set price(value:number){
        this._price = value;    
    } 
}